crucible-llvm-syntax/test-data/ptr.cbl

```diff
@@ -0,0 +1,24 @@
+(defun @test-ptr () (Ptr 64)
+  (start start:
+    (let blk0 (the Nat 0))
+    (let off0 (bv 64 0))
+    (let p0 (ptr 64 blk0 off0))
+    (let p (ptr-ite 64 #t p0 p0))
+    (let blk (ptr-block 64 p))
+    (let off (ptr-offset 64 p))
+    (assert! (equal? blk0 blk) "block numbers equal")
+    (assert! (equal? off0 off) "offsets equal")
```

Comment:
This test case makes me wonder: should we also add a `ptr-eq` statement as well? This would avoid having to manually check both the block and offset values each time.

Reply:
Yes! I've covered just some of the constructors in `LLVM.Extension.Syntax`, but ideally we'd do all of them, and `LLVM_PtrEq` is one of them.

But also, this test is basically just there to ensure we can parse each of these forms as expected, and I thought it'd be nice to have somewhat reasonable runtime behavior that actually checks *something*

Reply:
I'll make a follow-up issue to cover the remaining constructors once this is merged.